scheduler/planner.go

```diff
@@ -281,13 +260,74 @@ func (unit *Unit) RankValue() int64 {
 	// Increase the value for the number of dependencies, so that
 	// tasks (and units) which block other tasks run before tasks
 	// that don't block other tasks.
-	unit.cachedValue += priority * (numDeps / length)
+	value += priority * (u.NumDeps / length)
 
 	// Increase the value for tasks with longer runtimes, given
 	// that most of our workloads have different runtimes, and we
 	// don't want to have longer makespans if longer running tasks
 	// have to execute after shorter running tasks.
-	unit.cachedValue += priority * unit.distro.GetExpectedRuntimeFactor() * int64(math.Floor(expectedRuntime.Minutes()/float64(length)))
+	value += priority * u.Settings.GetExpectedRuntimeFactor() * int64(math.Floor(u.ExpectedRuntime.Minutes()/float64(length)))
+
+	return value
+}
+
+func (unit *Unit) info() unitInfo {
+	info := unitInfo{
+		Settings: unit.distro.PlannerSettings,
+	}
+
+	for _, t := range unit.tasks {
+		if t.Requester == evergreen.MergeTestRequester {
+			info.InCommitQueue = true
+		} else if evergreen.IsPatchRequester(t.Requester) {
+			info.InPatch = true
+		}
+
+		if t.TaskGroup == "" {
+			info.AnyNonGroupTasks = true
+		}
+		if t.GenerateTask {
+			info.GenerateTask = true
```

Comment:
nit: some of these if-and-then-set-true blocks could be simplified down (e.g. `info.GenerateTask = t.GenerateTask`).

Reply:
I refactored to use the value of t.GenerateTask